Class/Warrior/Spells/SunderArmor.cpp

```diff
@@ -0,0 +1,81 @@
+#include "SunderArmor.h"
+
+#include "Buff.h"
+#include "CastComplete.h"
+#include "CharacterStats.h"
+#include "CombatRoll.h"
+#include "CooldownControl.h"
+#include "HeroicStrike.h"
+#include "MainhandAttack.h"
+#include "OffhandAttack.h"
+#include "Warrior.h"
+#include "WarriorSpells.h"
+
+SunderArmor::SunderArmor(Warrior* pchar, WarriorSpells* spells) :
+    // TODO Find the right spell icon
+    Spell("SunderArmor", "Assets/ability/Ability_warrior_decisivestrike.png",
+          pchar, new CooldownControl(pchar, 0.0), RestrictedByGcd::Yes,
+          ResourceType::Rage,
+          15),
+    TalentRequirer(QVector<TalentRequirerInfo*>{new TalentRequirerInfo("Improved Sunder Armor", 3, DisabledAtZero::No)}),
+    warr(pchar),
+    spells(spells),
+    // TODO find these lower level values
+    innate_threat_by_spell_rank({0, 0, 0, 0, 260 }),
+    innate_threat(260),
+    talent_ranks({15, 14, 13, 12})
+{
+}
+
+SunderArmor::~SunderArmor() {
+    delete cooldown;
+}
+
+void SunderArmor::increase_talent_rank_effect(const QString&, const int curr) {
+    resource_cost = talent_ranks[curr];
+}
+
+void SunderArmor::decrease_talent_rank_effect(const QString&, const int curr) {
+    resource_cost = talent_ranks[curr];
+}
+
+SpellStatus SunderArmor::is_ready_spell_specific() const {
+    if (warr->in_battle_stance())
+      return SpellStatus::InBattleStance;
+
+    if (warr->in_berserker_stance())
+      return SpellStatus::InBerserkerStance;
```

Comment:
Remove the restrictions on stances above as Sunder Armor is usable in all stances. The only thing it should check is stance cooldown (the line on the return should remain as is).

Reply:
Oh wow, you're right. Thought it was defensive only.